secret_provider/src/types.rs

```diff
@@ -0,0 +1,34 @@
+use serde::{Deserialize, Serialize};
+
+/// Encryption parameters.
+#[derive(Serialize, Deserialize, Default, Debug, Clone, Eq, PartialEq)]
```

Comment:
Reviewers: We should use this type by wrapping it in a newtype in the control-plane and data-plane, rather than duplicating. Is that a fair idea?

Reply:
hmm how about decoupling the types from the `SecretProvider`? 
So basically `SecretProvider` will only know how to return a given type which implements deserialize, and then another abstraction can convert to the type we want.
Then we can have these types be the default, but also makes it easy to use a different type should we need to.

Reply:
Done.